.github/actions/setup-java-build/toolchains.xml

```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<toolchains>
+	xmlns="http://maven.apache.org/TOOLCHAINS/1.1.0"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/TOOLCHAINS/1.1.0 http://maven.apache.org/xsd/toolchains-1.1.0.xsd">
+	<toolchain>
+		<type>jdk</type>
+		<provides>
+			<version>17</version>
+		</provides>
+		<configuration>
+			<jdkHome>${env.JAVA_HOME}</jdkHome>
+		</configuration>
+	</toolchain>
+</toolchains>
```

Comment:
AFAIK maven should already autodetect the default JDK by using JAVA_HOME and we never had to configure a toolchains.xml file also when using different java versions in our existing builds

which is the actual issue you were facing that lead to this change? can you provide a branch with a build failing due to the lack of this file?

Reply:
Thanks for being super clear!

I understand that you had this issue because the parent project you are using (maintained by a third party) uses the [maven-toolchain-plugin](https://maven.apache.org/plugins/maven-toolchains-plugin/) which requires a toolchains.xml setup to not fail.

IMHO the `maven-toolchains-plugin` is not so commonly used in these days and I literally cannot find any usage in the entire alfresco codebase, hence I would prefer to not increase the complexity of our `setup-java-build` action to cover this not really common use case.

I suggest that you provide it directly inside the workflows of the builds that requires it, and only if you need to handle it in the same way in different repositories, then you can define your own action and reuse it within your different repositories (ping me in chat if you need more guidance on this)